monasca/templates/agent-daemonset.yaml

```diff
@@ -23,6 +23,22 @@ spec:
           resources:
 {{ toYaml .Values.agent.resources | indent 12 }}
           env:
+            - name: OS_AUTH_URL
+            {{- if .Values.agent.keystone.url }}
+            {{- if eq (kindOf .Values.agent.keystone.url) "map" }}
+              valueFrom:
+                secretKeyRef:
+                  name: "{{ .Values.agent.keystone.url.secret_name }}"
+                  key: "{{ .Values.agent.keystone.url.secret_key | default "OS_AUTH_URL" }}"
+            {{- else }}
+              value: "{{ .Values.agent.keystone.url }}"
```

Comment:
It's pretty annoying, yeah. I really wanted to make it a part of the template but we have a configurable default value, meaning the template needs access to `.Values`. Helm lets you pass exactly one variable to a template as its scope, so either we give it all of `.Values` - and it has no idea where to look for the auth for this specific component - or we give it `.Values.agent.auth` and it has no way to get the default keystone URL.

I'd hoped helm's [variables](https://github.com/kubernetes/helm/blob/master/docs/chart_template_guide/variables.md) would have been a workaround, but they don't seem to work in `{{ include }}` calls :frowning_face: